### PR TITLE
fix(card): Improve card styles for react

### PR DESCRIPTION
**Pull request by clenemt**
:arrow_forward: **Related to**: BLOS-109
:eyeglasses: **Context**: these changes are needed for the react version of this component.
:sunrise: **After**:

### CODE REVIEW

File: src/components/_card.scss [hunk 1]
```diff
@@ -8,20 +8,19 @@
...
 //   <div class="sb-card__header">
-//     <h1 class="sb-card__title sb-title sb-text--center">A very long and boring card title that can span over and over</h1>
+//     <h1 class="sb-title sb-text--center">A very long and boring card title that can span over and over</h1>
 //   </div>
 //   <div class="sb-card__body">
-//     <p class="sb-text sb-text--center sb-text--number sb-text--uppercase sb-text--emphasis">221 responses</p>
-//     <p class="sb-text--s sb-text--center sb-text--subtle">3 data sources</p>
+//     <button class="sb-btn sb-btn--primary">Click me</button>
 //   </div>
 // </a>
```
Comment (spirosikmd) [inline src/components/_card.scss]: Shouldn't this be a closing `div` tag?

File: src/components/_card.scss [hunk 2]
```diff
@@ -38,25 +37,26 @@
...
-  }
-
-  &:hover,
-  &:focus {
-    outline: 0;
-    border: 1px solid $sb-color-blue-2;
-    transform: translate(0, -3px, 0);
+a.#{$ns}card:hover,
+a.#{$ns}card:focus,
+.#{$ns}card:focus-within {
```
Comment (spirosikmd) [inline src/components/_card.scss]: Why are we applying `hover` and `focus` on the `a` tag explicitly and `focus-within` only to the class?

File: src/components/_card.scss (hunk 1 above)
Comment (clenemt) [reply src/components/_card.scss]: Noice catch 👍

File: src/components/_card.scss (hunk 2 above)
Comment (clenemt) [reply src/components/_card.scss]: Because a card is only interactive (transition and border color) when it is an anchor or there is a focusable content inside

File: src/components/_card.scss [hunk 3]
```diff
@@ -38,25 +37,26 @@
   border-radius: sb-px2rems($sb-radius-l);
   box-shadow: 0 2px 8px 0 rgba($sb-color-dark-2, 0.1);
   transition: all $sb-transition;
+
+  .#{$ns}title {
```
Comment (dine) [inline src/components/_card.scss]: why did you move away from styling `#{$ns}card__title` to just `#{$ns}title`? I think it is confusing to have styles from `#{$ns}title` inside the `_card.scss` file (and you also commented something similar here) https://github.com/usabilla/stylabilla/pull/193/files#r210531310

File: src/components/_card.scss [hunk 4]
```diff
@@ -38,25 +37,26 @@
...
+a.#{$ns}card:focus,
+.#{$ns}card:focus-within {
+  outline: 0;
+  border: 1px solid $sb-color-blue-2;
+  transform: translate(0, -3px, 0);
 
-    .#{$ns}card__title {
-      color: $sb-color-blue-2;
-    }
+  .#{$ns}title {
```
Comment (dine) [inline src/components/_card.scss]: same as before

File: src/components/_card.scss (hunk 3 above)
Comment (clenemt) [reply src/components/_card.scss]: Because it is easier when building the react component.
If I force us to add the `card__title`:
* when we pass a node to the `header="..."`, we have to add the class `card__title` every time which is very uggly

Comment (spirosikmd) [reply src/components/_card.scss]: But isn't it the same? As now you need to add the class `sb-title` anyway when you pass a node?

Comment (clenemt) [reply src/components/_card.scss]: Yea it is the same, but I feel better doing this:

```jsx
<Card header={ <h1 class="sb-title">foo</h1> } />
```

then this: 

```jsx
<Card header={ <h1 class="sb-card__title sb-title">foo</h1> } />
```

Other alternatives would be:

1- force the color override of the `sb-title` with:
```js
.#{$ns}card__header > * {
  color: $sb-color-blue-2;
}
```

which would work even if their is no `sb-title`

2- not use the `sb-title` and instead extend the `sb-card__title`:
```css
.#{$ns}card__title {
  @include sb-title();
}
```

```jsx
<Card header={ <h1 class="sb-card__title">foo</h1> } />
```

**Comment by clenemt** [comment]
I fixed the stylelint plugin so we can now have regular comments in there. Please have a look again @dine @spirosikmd

File: lint-docs-required.js [hunk 5]
```diff
@@ -1,44 +1,64 @@
...
+  generic: 'File is missing kss documentation',
+});
+
+module.exports = stylelint.createPlugin(ruleName, actual => (root, result) => {
+  const validOptions = stylelint.utils.validateOptions(result, ruleName, { actual });
+  if (!validOptions) return;
+
+  let commentLines = [];
+  let hasDocTexts = false;
+  let isFileParsed = true;
```
Comment (spirosikmd) [inline lint-docs-required.js]: As far as I can see, this never becomes `false`. What is the purpose of this variable?

File: lint-docs-required.js [hunk 6]
```diff
@@ -1,44 +1,64 @@
...
+
+  root.walkComments(comment => {
+    isFileParsed = true;
+
+    const next = comment.next() && comment.next().text;
```
Comment (spirosikmd) [inline lint-docs-required.js]: Can we have this check `if (next !== undefined) return` here and return early? Or do we still need to add the `line` in the `commentLines` even if `next` is not undefined? If we still need it, maybe we can change the order of things?

```js
const line = comment.text;
if (line.includes('stylelint-disable')) return;

const next = comment.next() && comment.next().text;
if (next !== undefined) return;

// Then continue with the rest
const text = commentLines.join(' '); 
// ...
```

File: lint-docs-required.js [hunk 7]
```diff
@@ -1,44 +1,64 @@
...
+    const line = comment.text;
+
+    // Disregard this single line if there is a `stylelint-disable` text in there
+    if (line.includes('stylelint-disable')) return;
+    commentLines.push(line);
+
+    // Process the comment once we actually have all the lines
+    if (next === undefined) {
+      const text = commentLines.join(' ');
+      const hasDocText = requiredDocTexts.some(docText => text.includes(docText));
```
Comment (spirosikmd) [inline lint-docs-required.js]: Can't we return directly after this line with `if (!hasDocText) return;`?

File: lint-docs-required.js (hunk 5 above)
Comment (clenemt) [reply lint-docs-required.js]: Sometime a file has eslint rule disabled at top level, hence it does not get parsed

File: lint-docs-required.js (hunk 6 above)
Comment (clenemt) [reply lint-docs-required.js]: We need to add the line for the comment as we are parsing each line and adding them to the array until we reach the end of the comment block. But I have removed the if indentation.

File: lint-docs-required.js (hunk 7 above)
Comment (clenemt) [reply lint-docs-required.js]: Nope as we need to reset the `commentLines`. But I have move the logic around for easier read.

File: lint-docs-required.js [hunk 8]
```diff
@@ -1,44 +1,66 @@
...
+  root.walkComments(comment => {
+
+    const next = comment.next() && comment.next().text;
+    const line = comment.text;
+
+    // Disregard this single line comment if there is a `stylelint-disable` text in there
+    if (!line || line.includes('stylelint-disable')) return;
+    isFileParsed = true;
+
+    console.log(comment.source.input.file);
```
Comment (spirosikmd) [inline lint-docs-required.js]: Do we need this `console.log` here?

File: src/trumps/_helpers.scss [hunk 9]
```diff
@@ -1,3 +1,17 @@
+// Visibility Classes
+//
+// Some visibility helper classes.
+//
+// Browser support: IE9+, Safari6+, Firefox30+, Chrome30+
+//
+// Markup:
+// <div class="sb-tile sb-padding {{modifier_class}}">I'm visible</div>
```
Comment (spirosikmd) [inline src/trumps/_helpers.scss]: Typo: `sb-title`
